Tasks/DeployVisualStudioTestAgent/TestAgentInstall.ps1

```diff
@@ -37,6 +37,42 @@ function Install-Product($SetupPath, $UserName, $Password, $ProductVersion, $Arg
 			Write-Warning -Verbose "Caught exception while installing Test Agent"
 			throw $_.Exception
 		}
+                 
+		if($exitCode -eq -2147185721)
+		{
+			# pending restart .
+			try
+			{
+				$testAgentFile = "$env:SystemDrive\TestAgent\testagent.txt"
+				$testAgentFileExists = Test-Path $testAgentFile
+				if($testAgentFileExists)
+				{
+					# delete the file which indicated that test agent installation failed.
+					remove-item $testAgentFile | Out-Null
+					# we have retried once .Now fail with appropriate message
+					Write-Warning -Verbose "Retried to install Test Agent"
```

Comment:
My theory beyond this is  retrying is not error but the failing with some error code is error which we are throwing as error. But before giving that message I should somewhere indicate that we have retried. Instead of giving just as info, warning would grab the attention. It is between the message and error message.